OsmAnd/src/net/osmand/plus/myplaces/FavoritesActivity.java

```diff
@@ -187,13 +193,39 @@ private void setTabs(List<TabItem> mTabs) {
 
 	@Override
 	public void onAttachFragment(Fragment fragment) {
-		fragList.add(new WeakReference<>(fragment));
+		if (fragment instanceof FavoritesFragmentStateHolder) {
+			if (intentParams != null && intentParams.getInt(TAB_ID, -1) != -1) {
+				Bundle b = new Bundle();
+				int tabId = intentParams.getInt(TAB_ID, FAV_TAB);
+				if (tabId == FAV_TAB && fragment instanceof FavoritesTreeFragment) {
+					b.putString(FavoritesFragmentStateHolder.GROUP_NAME_TO_SHOW,
```

Comment:
Use OOP. 
all the logic of putInt, getInt should be hidden behind FavoritesFragmentStateHolder.

NotesFragment belongs to Plugin so it couldn't be used here !!!

Reply:
OK, think I get it. Will do.